### ISSUE TITLE
It is recommended to add the automatic elimination logic after hanging up for more than 30 minutes

### ISSUE THREAD

**Issue by sunl888** [ddnet/ddnet#3968]
It is recommended to add the automatic elimination logic after hanging up for more than 30 minutes, which always prompts that the server is full. In fact, there are a lot of players hanging up

**Comment by Learath2** [comment]
Hm, do you mean the timeout protection duration? I think we lowered that already

**Comment by sunl888** [comment]
> Hm, do you mean the timeout protection duration? I think we lowered that already

However, the actual situation is that many people hang up inside and last for a long time, resulting in other players unable to enter the game normally at the peak of the game, such as kobra. I mean, can you appropriately reduce the duration of hibernation during the peak period of the game or in the popular map?

**Comment by murpii** [comment]
I think hes refering to the people afking for multiple hours

**Comment by Learath2** [comment]
We could look into re-enabling afk kicking, especially on fuller servers

**Comment by Astramast** [comment]
Yeah so basically i agree. If a player is ruled as afk by the server and the server is full, this player should be kicked for inactivity. There's too much people just doing nothing on full servers with people wanting to play, and joining an empty server with the same map results generally in you playing alone. So the best option is just to kick inactive players one by one, each time the server is full (So each time that someone connects to the server, if the server is now full -> kick the player being afk for the longest time if it exists) until there's no afk player anymore to kick. Then start a queue would be the best thing for players so they could know how much people are waiting to enter. (I think its better than trying to connect every 4s or just spam-refreshing the main server page like i do 🙃 ) The player queue is an already open discussion : https://github.com/ddnet/ddnet/issues/3453
